src/components/ValuePicker/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, {forwardRef, useState} from 'react';
+import type {ForwardedRef} from 'react';
+import {View} from 'react-native';
+import FormHelpMessage from '@components/FormHelpMessage';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import type {ValuePickerItem, ValuePickerProps} from './types';
+import ValueSelectorModal from './ValueSelectorModal';
+
+function ValuePicker({value, label, items, placeholder = '', errorText = '', onInputChange, furtherDetails, shouldShowTooltips = true}: ValuePickerProps, forwardedRef: ForwardedRef<View>) {
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+
+    const showPickerModal = () => {
+        setIsPickerVisible(true);
+    };
+
+    const hidePickerModal = () => {
+        setIsPickerVisible(false);
+    };
+
+    const updateInput = (item: ValuePickerItem) => {
+        if (item?.value && item.value !== value) {
+            onInputChange?.(item.value);
+        }
+        hidePickerModal();
+    };
+
+    const descStyle = value?.length === 0 ? StyleUtils.getFontSizeStyle(variables.fontSizeLabel) : null;
```

Comment:
This condition doesn't look the same as before `value?.length === 0`
What if value is `undefined`
`value?.length === 0` will give `false`
while old condition 
`!value || value.length === 0` gives `true`

Reply:
Right, reverted